### ISSUE TITLE
Bun.write with `new Response(req.body)` hangs

### ISSUE THREAD

**Issue by kindlyfire** [oven-sh/bun#13237]
### What version of Bun is running?

1.1.22

### What platform is your computer?

Linux 6.9.3-76060903-generic x86_64 x86_64

### What steps can reproduce the bug?

Execute the following code:
```ts
const server = Bun.serve({
    async fetch(req) {
        await Bun.write('test.txt', new Response(req.body))
        return new Response('ok')
    },
})
console.log(server.url.href)
```
And then make a POST request to the server. The request never completes, and `Bun.write` hangs indefinitely.

### What is the expected behavior?

The contents of the request being correctly written to the file.

### What do you see instead?

`Bun.write` hangs, and the request never resolves.

**Comment by Hamza12700** [comment]
Yes, it does hang indefinitely. I don't know if this issue is in `Bun.serve` or `Bun.write`.
In there recent blog they said they have made some improvement in [Bun.serve](https://bun.sh/blog/bun-v1.1.22#10-higher-throughput-post-requests-in-bun-serve) to have 10% higher throughput POST request if the code doesn't read the body of an incoming request.

**Comment by cirospaciari** [comment]
Investigating the issue looks like the options bellow works fine:
```js
const server = Bun.serve({
    async fetch(req) {
        await Bun.write('test.txt', req);
         await Bun.write("test.txt", await Bun.readableStreamToArrayBuffer(req.body));
        await Bun.write("test2.txt", await fetch("https://example.com"));
        await Bun.write("test.txt", new Response("Text here..."));

        return new Response('ok')
    },
})
console.log(server.url.href)
```

**Comment by Hamza12700** [comment]
Okay, then why does it hang like this?
```js
const server = Bun.serve({
    async fetch(req) {
        await Bun.write('test.txt', new Response(req.body));
        return new Response('ok')
    },
})
console.log(server.url.href)
```

**Comment by cirospaciari** [comment]
> Okay, then why does it hang like this?
> 
> ```js
> const server = Bun.serve({
>     async fetch(req) {
>         await Bun.write('test.txt', new Response(req.body));
>         return new Response('ok')
>     },
> })
> console.log(server.url.href)
> ```

Still investigating the issue, but will give a feedback when found out.

**Comment by empz** [comment]
I'm also experiencing this issue. I'm using Bun's fetch to request a file from another server in the network and sometimes writing the response out to the file system hangs forever...

```ts
const response = await Bun.fetch(url, {
	method: "POST",
	body: JSON.stringify(args),
	headers: {
		"Content-Type": "application/json",
	},
	// @ts-expect-error
	timeout: false,
});

if (!response.ok) {
	throw new Error("Fetch failed");
}

const bytesWritten = await Bun.write(outputLocation, response);
```

Is there a workaround for this while the issue is being investigated?

**Comment by belizariogr** [comment]
Same error here... the workaround that i made was save to a blob before write:

` const img = await response.blob()
        await Bun.write(savePath + arquivo.IdPublico, img);`

**Comment by jahudka** [comment]
Workaround which works and doesn't need to buffer the entire file in memory:

```typescript
import { open } from 'node:fs/promises';

export async function write(path: string, file: Response): Promise<number> {
  if (!file.body) {
    return 0;
  }

  const reader = file.body.getReader();
  const fp = await open(path, 'w', 0o640);
  let size = 0;

  for (let res = await reader.read(); !res.done; res = await reader.read()) {
      const { bytesWritten } = await fp.write(res.value);
      size += bytesWritten;
  }

  await fp.close();
  return size;
}
```

**Comment by empz** [comment]
Is this still an issue in v1.2?

**Comment by jahudka** [comment]
@empz yeah I'm on 1.2 - I actually only just started out with Bun, so I don't have experience with pre-1.2 behaviour

**Comment by miku-orb** [comment]
With Bun v1.2.14 (and some earlier versions), it **randomly** hangs at `Bun.write` below (but not always!). Retrying the fetch after killing Bun process works. Not sure if relevant to this issue (which seems to claim a problem with it *always* hanging).

```ts
await fetch("http://example.com/test?id=" + id, {
    method: "GET",
}).then(r => {
    // here a header is read with `r.headers.get`, if it matters
    console.log(`Fetched ${id}`); // this gets logged, so we know a fetch didn't hang
    return r;
}).then(r => Bun.write(`./${id}`, r));
```

But indeed by changing the last line to something like:
```ts
.then(async r => Bun.write(`./${id}`, await r.blob()));
```
...it seems to help, no hangs since then (the only problem is that I get a TypeScript error in my IDE for the second argument, but it executes fine)

**Comment by david-arteaga** [comment]
I've had it sometimes hangs when trying to download files:
```ts
await Bun.write(filePath, await fetch(downloadUrl))
```

I'm using `Bun.serve` to serve a full stack react app.

I have to download 100s of small (100-400kb) files and it hangs pretty consistently (even when I try to download batches of 5, it hands on the second batch).

Not really sure what is going on here, but this seems like a pretty egregious bug 🤔 .
